api/client/client.go

```diff
@@ -613,6 +613,66 @@ func (c *Client) DevicesClient() devicepb.DeviceTrustServiceClient {
 	return devicepb.NewDeviceTrustServiceClient(c.conn)
 }
 
+// CreateDeviceResource creates a device using its resource representation.
+// Prefer using [DevicesClient] directly if you can.
+func (c *Client) CreateDeviceResource(ctx context.Context, res *types.DeviceV1) (*types.DeviceV1, error) {
+	dev, err := types.DeviceFromResource(res)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	created, err := c.DevicesClient().CreateDevice(ctx, &devicepb.CreateDeviceRequest{
+		Device:           dev,
+		CreateAsResource: true,
+	}, c.callOpts...)
+	if err != nil {
+		return nil, trail.FromGRPC(err)
+	}
+	return types.DeviceToResource(created), nil
```

Comment:
The generator ignores return value `_, err := r.p.Client.CreateDeviceResource(ctx, &types.DeviceV1{})` and expects `res *types.DeviceV1` to be updated.
Defined here: https://github.com/gravitational/teleport-plugins/blob/master/terraform/_gen/main.go#L43

Reply:
Discussed a bit offline: let's see if we can sort this out in the generator side. Changing input params isn't expected from methods like this, and it's not very practical with protos either. Protos have a "DoNotCopy" embed, so stuff like `*res = *created` won't work, and copying all fields manually will add another maintenance burden.